java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/Server.java

```diff
@@ -1049,44 +1051,53 @@ private void collectProjectCandidates(FileObject fo, List<FileObject> candidates
 
         private void initializeOptions() {
             getWorkspaceProjects().thenAccept(projects -> {
-                ConfigurationItem item = new ConfigurationItem();
-                // PENDING: what about doing just one roundtrip to the client- we may request multiple ConfiguratonItems in one message ?
-                item.setSection(client.getNbCodeCapabilities().getConfigurationPrefix() + NETBEANS_JAVA_HINTS);
-                client.configuration(new ConfigurationParams(Collections.singletonList(item))).thenAccept(c -> {
-                    if (c != null && !c.isEmpty() && c.get(0) instanceof JsonObject) {
-                        textDocumentService.updateJavaHintPreferences((JsonObject) c.get(0));
-                    }
-                    else {
-                        textDocumentService.hintsSettingsRead = true;
-                        textDocumentService.reRunDiagnostics();
-                    }
-                });
-                item.setSection(client.getNbCodeCapabilities().getConfigurationPrefix() + NETBEANS_PROJECT_JDKHOME);
-                client.configuration(new ConfigurationParams(Collections.singletonList(item))).thenAccept(c -> {
-                    JsonPrimitive newProjectJDKHomePath = null;
+                List<String> defaultConfigs = List.of(NETBEANS_JAVA_HINTS, NETBEANS_PROJECT_JDKHOME);
+                List<String> projectConfigs = List.of(NETBEANS_FORMAT, NETBEANS_JAVA_IMPORTS);
 
-                    if (c != null && !c.isEmpty() && c.get(0) instanceof JsonPrimitive) {
-                        newProjectJDKHomePath = (JsonPrimitive) c.get(0);
-                    } else {
-                    }
-                    textDocumentService.updateProjectJDKHome(newProjectJDKHomePath);
-                });
+                AtomicReference<FileObject> projectDirectory = new AtomicReference<>(null);
```

Comment:
The use of `AtomicReference` here is probably unnecessary, correct? Can it be simply `FileObject projectDirectory`? It seems to be assign on one place only.